test-framework/junit5-component/src/test/java/io/quarkus/test/component/declarative/InstanceInterfaceTest.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.test.component.declarative;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import jakarta.enterprise.inject.Instance;
+import jakarta.inject.Inject;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.component.QuarkusComponentTest;
+
+@QuarkusComponentTest
+public class InstanceInterfaceTest {
```

Comment:
I think we should also add a test that injects an interface but has some impl that we don't remove because of it.
So something like this:

```
@QuarkusComponentTest
public class ListAllInterfaceWithImplsTest {

    @Inject
    @All
    List<SomeInterface> components;

    @Test
    public void testComponents() {
        // SomeInterface is registered as component but no implementation is added automatically
        assertEquals(1, components.size());
    }

    @Dependent
    public static class FooBean implements SomeInterface {

    }
}
```

Reply:
Done.